homeassistant/components/isy994/binary_sensor.py

```diff
@@ -28,83 +65,187 @@ def setup_platform(
+                # reported as "10", which translate from Hex to 10 and 16 resp.
+                device = ISYInsteonBinarySensorEntity(
+                    node, DEVICE_CLASS_PROBLEM, initial_state
+                )
+                devices.append(device)
+                continue
+            if subnode_id in SUBNODE_MOTION_DISABLED:
+                # Motion Disabled Sub-node for MS II ("D" or "13")
+                device = ISYInsteonBinarySensorEntity(node)
+                devices.append(device)
+                continue
+
+        # We don't yet have any special logic for other sensor
+        # types, so add the nodes as individual devices
+        device = ISYBinarySensorEntity(node, device_class)
+        devices.append(device)
 
     for name, status, _ in hass.data[ISY994_PROGRAMS][BINARY_SENSOR]:
         devices.append(ISYBinarySensorProgramEntity(name, status))
 
     add_entities(devices)
 
 
-def _detect_device_type(node) -> str:
+def _detect_device_type_and_class(node: Union[Group, Node]) -> (str, str):
     try:
         device_type = node.type
     except AttributeError:
         # The type attribute didn't exist in the ISY's API response
-        return None
+        return (None, None)
+
+    # Z-Wave Devices:
+    if node.protocol == PROTO_ZWAVE:
+        device_type = f"Z{node.zwave_props.category}"
+        for device_class in [*BINARY_SENSOR_DEVICE_TYPES_ZWAVE]:
+            if (
+                node.zwave_props.category
+                in BINARY_SENSOR_DEVICE_TYPES_ZWAVE[device_class]
+            ):
+                return device_class, device_type
+        return (None, device_type)
+
+    # Other devices (incl Insteon.)
+    for device_class in [*BINARY_SENSOR_DEVICE_TYPES_ISY]:
+        if any(
+            [
+                device_type.startswith(t)
+                for t in set(BINARY_SENSOR_DEVICE_TYPES_ISY[device_class])
+            ]
+        ):
+            return device_class, device_type
+    return (None, device_type)
 
-    split_type = device_type.split(".")
-    for device_class, ids in ISY_BIN_SENS_DEVICE_TYPES.items():
-        if f"{split_type[0]}.{split_type[1]}" in ids:
-            return device_class
 
-    return None
+class ISYBinarySensorEntity(ISYNodeEntity, BinarySensorEntity):
+    """Representation of a basic ISY994 binary sensor device."""
 
+    def __init__(self, node, force_device_class=None, unknown_state=None) -> None:
+        """Initialize the ISY994 binary sensor device."""
+        super().__init__(node)
+        self._device_class = force_device_class
 
-class ISYBinarySensorEntity(ISYNodeEntity, BinarySensorEntity):
-    """Representation of an ISY994 binary sensor device.
+    @property
+    def is_on(self) -> bool:
+        """Get whether the ISY994 binary sensor device is on.
+
+        Note: This method will return false if the current state is UNKNOWN
+        """
+        return bool(self.value)
+
+    @property
+    def state(self):
```

Comment:
This is existing so can be cleaned up later.

Binary sensors are only supposed to have two states.  These should only implement `is_on` and not `state`
https://developers.home-assistant.io/docs/entity_binary_sensor/

If the state is `unknown` https://developers.home-assistant.io/docs/entity_index#generic-properties we should set `available` to False